torch/lib/c10d/test/FileStoreTest.cpp

```diff
@@ -0,0 +1,120 @@
+#include <unistd.h>
+
+#include <condition_variable>
+#include <iostream>
+#include <mutex>
+#include <sstream>
+#include <thread>
+
+#include "FileStore.hpp"
+
+using namespace c10d;
+
+class Semaphore {
+ public:
+  void post(int n = 1) {
+    std::unique_lock<std::mutex> lock(m_);
+    n_ += n;
+    cv_.notify_all();
+  }
+
+  void wait(int n = 1) {
+    std::unique_lock<std::mutex> lock(m_);
+    while (n_ < n) {
+      cv_.wait(lock);
+    }
```

Comment:
nit: C++11 idiom for this is `cv_.wait_for(lock, [n, this]() { return this->n_ >= n; })`

Reply:
That signature also requires passing a timeout.

Reply:
My bad. Should have been only `.wait`. It's a minor thing, so ignore it.